# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RemoteOuterWindowProxy.cpp
# Commit: a30f158eba45
# Full Hash: a30f158eba4511313be684c4b5f3d5f278f34103
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-22 21:48:48
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 276b131190a8 -r a30f158eba45 dom/base/RemoteOuterWindowProxy.cpp
--- a/dom/base/RemoteOuterWindowProxy.cpp	Wed Apr 22 03:21:55 2020 +0000
+++ b/dom/base/RemoteOuterWindowProxy.cpp	Wed Apr 22 06:05:41 2020 +0000
@@ -95,7 +95,7 @@
   BrowsingContext* bc = GetBrowsingContext(aProxy);
   uint32_t index = GetArrayIndexFromId(aId);
   if (IsArrayIndex(index)) {
-    const BrowsingContext::Children& children = bc->GetChildren();
+    Span<RefPtr<BrowsingContext>> children = bc->Children();
     if (index < children.Length()) {
       return WrapResult(aCx, aProxy, children[index],
                         JSPROP_READONLY | JSPROP_ENUMERATE, aDesc);
@@ -120,7 +120,7 @@
       return false;
     }
 
-    for (BrowsingContext* child : bc->GetChildren()) {
+    for (BrowsingContext* child : bc->Children()) {
       if (child->NameEquals(str)) {
         return WrapResult(aCx, aProxy, child, JSPROP_READONLY, aDesc);
       }
@@ -132,7 +132,7 @@
 
 bool AppendIndexedPropertyNames(JSContext* aCx, BrowsingContext* aContext,
                                 JS::MutableHandleVector<jsid> aIndexedProps) {
-  int32_t length = aContext->GetChildren().Length();
+  int32_t length = aContext->Children().Length();
   if (!aIndexedProps.reserve(aIndexedProps.length() + length)) {
     return false;
   }